### PR TITLE
Add password input delay to slow down brute force attacks

### LINKED ISSUE #740: Password input delay to resist brute force attacks
On screens that require users to input their passwords or their pins, in the case of a wrong password/pin there should be a delay before the user is allowed to input a password/pin again.

The aim of this is to significantly slow down/resist brute force attacks

### PR DESCRIPTION
**Pull request by C-ollins**
This pr adds a 5 seconds delay after every even password trial and 2 seconds delay for otherwise in an attempt to slow down a brute force attack.

Closes #740 
(Password & keyboard are censored but the delay is visible)

### CODE REVIEW

**Comment by degeri** [comment]
ideally brute force protections should be at the logical layer. Perhaps increase number of derivation iterations based on phone specs ?

**Comment by C-ollins** [comment]
Yeah, we could do that but for now, we're leaving the number of derivation iterations to dcrwallet's default as it's not currently possible to change the number outside dcrwallet and we're currently investigating how the current derivation iterations could be increasing the app's CPU/memory usage.

**Review by JustinBeBoy — DISMISSED** [review]

File: Decred Wallet/Features/Custom Dialogs/SimpleTextInputDialog.swift [hunk 1]
```diff
@@ -138,6 +139,10 @@ extension SimpleTextInputDialog: InputDialogDelegate {
         self.submitButton.isEnabled = true
         self.submitButton.stopLoading()
     }
+    
+    func displayPassphraseError(errorMessage: String) {
```
[the remark this answers is not in the document]
Comment (C-ollins) [reply Decred Wallet/Features/Custom Dialogs/SimpleTextInputDialog.swift:143]:
```
Default argument not permitted in a protocol method
```
I get this error when I try to add an optional parameter to `InputDialogDelegate`.

[the remark this answers is not in the document]
Comment (C-ollins) [reply Decred Wallet/Features/Custom Dialogs/SimpleTextInputDialog.swift:143]: Nevermind.

**Review by dreacot — APPROVED** [review]
LGTM

File: Decred Wallet/Features/Security/SecurityCodeRequestBaseViewController.swift [hunk 2]
```diff
@@ -50,6 +50,9 @@ class SecurityCodeRequestBaseViewController: UIViewController {
     func hideError() {
         self.isInErrorState = false
     }
+    
+    func showPassphraseError(text: String) {
```
Comment (dreacot) [inline Decred Wallet/Features/Security/SecurityCodeRequestBaseViewController.swift:54]: @C-ollins nothing seems to be done inside this function

Comment (C-ollins) [reply Decred Wallet/Features/Security/SecurityCodeRequestBaseViewController.swift:54]: The function is implemented in subclasses.
https://github.com/planetdecred/dcrios/pull/749/files#diff-a03f6b93d054b97dd40df7affd98091d143090b230b94b6f916541ab3801683bR224

**Review by dreacot — APPROVED** [review]

**Review by macsleven — APPROVED** [review]

**Pull request merged**